pkg/operator/ceph/object/config.go

```diff
@@ -190,6 +190,12 @@ func (c *clusterConfig) generateMonConfigOptions(rgwConfig *rgwConfig) (map[stri
 		configOptions["rgw_enable_ops_log"] = "false"
 	}
 
+	if c.store.Spec.Gateway.RgwReadAffinity != nil {
+		if c.clusterInfo.CephVersion.IsAtLeastTentacle() {
+			configOptions[radosReadReplicaPolicy] = c.store.Spec.Gateway.RgwReadAffinity.Type
+		}
+	}
```

Comment:
We are obligated to unset `rados_read_replica_policy` when the user changes `readAffinity` from non-nil to nil (or at least set it to `"default"`) so that they don't end up with this being enabled after they have indicated they no longer want it. 

Do we know if RGW can change its behavior for `rados_read_replica_policy` at runtime? If so, setting it here is probably good, and we can set `"default"` if the section is nil. If it can't change at runtime, RGW will need restarted, and we should use a CLI flag instead.

```suggestion
	if c.store.Spec.Gateway.RgwReadAffinity != nil {
		if c.clusterInfo.CephVersion.IsAtLeastTentacle() {
			configOptions[radosReadReplicaPolicy] = c.store.Spec.Gateway.RgwReadAffinity.Type
		}
	} else {
		// TODO: unset rados_read_replica_policy when read affinity is non-nil
	}
```

Reply:
> Do we know if RGW can change its behavior for rados_read_replica_policy at runtime

RGW needs to be restarted.

Reply:
This PR comment has not been addressed so far. Rest comments have been addressed.

Reply:
Now using setting the readAffinity via the container args.  Below are the logs for `localize` reads

```
Defaulted container "rgw" out of: rgw, chown-container-data-dir (init)
+ hostName=minikube-test-1
+ exec radosgw --keyring=/etc/ceph/keyring-store/keyring --default-log-to-stderr=true --default-err-to-stderr=true --default-mon-cluster-log-to-stderr=true '--default-log-stderr-prefix=debug ' --default-log-to-file=false --default-mon-cluster-log-to-file=false '--mon-host=[v2:10.111.246.71:3300,v1:10.111.246.71:6789]' --mon-initial-members=a --id=rgw.my.store.a --setuser=ceph --setgroup=ceph --foreground '--rgw-frontends=beast port=8080' --host=rook-ceph-rgw-my-store-a-8467984694-g8n8b --rgw-mime-types-file=/etc/ceph/rgw/mime.types --rgw-realm=my-store --rgw-zonegroup=my-store --rgw-zone=my-store --rados-replica-read-policy=localize '--crush-location=host=$hostName'
debug 2025-03-27T16:17:04.392+0000 7f1f43564940  0 deferred set uid:gid to 167:167 (ceph:ceph)
debug 2025-03-27T16:17:04.392+0000 7f1f43564940  0 ceph version 20.0.0-815-ge90047cc (e90047cc6981964461dbd2557d6fe7280e5f5cd1) tentacle (dev), process radosgw, pid 1
debug 2025-03-27T16:17:04.392+0000 7f1f43564940  0 framework: beast
debug 2025-03-27T16:17:04.393+0000 7f1f43564940  0 framework conf key: port, val: 8080
debug 2025-03-27T16:17:04.393+0000 7f1f43564940  1 init_numa not setting numa affinity
debug 2025-03-27T16:17:04.997+0000 7f1f43564940  0 framework: beast
debug 2025-03-27T16:17:04.997+0000 7f1f43564940  0 framework conf key: ssl_certificate, val: config://rgw/cert/$realm/$zone.crt
debug 2025-03-27T16:17:04.997+0000 7f1f43564940  0 framework conf key: ssl_private_key, val: config://rgw/cert/$realm/$zone.key
debug 2025-03-27T16:17:05.145+0000 7f1f43564940  0 starting handler: beast
debug 2025-03-27T16:17:05.249+0000 7f1f43564940  0 set uid:gid to 167:167 (ceph:ceph)
debug 2025-03-27T16:17:05.250+0000 7f1f43564940  1 mgrc service_daemon_register rgw.4965 metadata {arch=x86_64,ceph_release=tentacle,ceph_version=ceph version 20.0.0-815-ge90047cc (e90047cc6981964461dbd2557d6fe7280e5f5cd1) tentacle (dev),ceph_version_short=20.0.0-815-ge90047cc,container_hostname=rook-ceph-rgw-my-store-a-8467984694-g8n8b,container_image=quay.ceph.io/ceph-ci/ceph:main,cpu=Intel(R) Xeon(R) Silver 4114 CPU @ 2.20GHz,distro=centos,distro_description=CentOS Stream 9,distro_version=9,frontend_config#0=beast port=8080,frontend_type#0=beast,hostname=minikube.test.1,id=my.store.a,kernel_description=#1 SMP Thu May 9 02:07:35 UTC 2024,kernel_version=5.10.207,mem_swap_kb=0,mem_total_kb=5923924,num_handles=1,os=Linux,pid=1,pod_name=rook-ceph-rgw-my-store-a-8467984694-g8n8b,pod_namespace=rook-ceph,realm_id=edd1ad3c-3416-493d-ba5a-2dc75fd2a2fb,realm_name=my-store,zone_id=09e2e820-43ce-4eea-a364-824c7a652de2,zone_name=my-store,zonegroup_id=e4978c5a-6f27-4dd5-8ecc-e4d56ed2b031,zonegroup_name=my-store}


sh-5.1$ ceph config show client.rgw.my.store.a | grep localize
rados_replica_read_policy       localize                                       cmdline
sh-5.1$

```